File: python/31a_1624A_PlusOneOnTheSubset.py
def main():
    ans = []
    t = int(input())
    for _ in range(t):
        maxi, mini = 0, 1000000001
        n = int(input())
        array = input().split()
        for i in range(n):
            array[i] = int(array[i])
            maxi = max(array[i], maxi)
            mini = min(array[i], mini)
        ans.append((maxi - mini))
    for i in ans:
        print(i)

if __name__ == "__main__":
    main()

File: cpp/31a_1624A_PlusOneOnTheSubset.cpp
#include<bits/stdc++.h>

int main(){
    int t, n, max, min, temp;
    std::vector<int> ans;
    std::cin>>t;
    for(int i = 0; i < t; i++){
        std::cin>>n;
        max = 0, min = 1000000001;
        for(int j = 0; j < n; j++){
            std::cin>>temp;
            min = std::min(min, temp);
            max = std::max(max, temp);
        }
        ans.push_back((max - min));
    }
    for(auto i : ans){
        std::cout<<i<<"\n";
    }
    return 0;
}